Sources/SwiftProtobuf/Message+FieldMask.swift

```diff
@@ -0,0 +1,126 @@
+// Sources/SwiftProtobuf/Message+FieldMask.swift - Message field mask extensions
+//
+// Copyright (c) 2014 - 2023 Apple Inc. and the project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See LICENSE.txt for license information:
+// https://github.com/apple/swift-protobuf/blob/main/LICENSE.txt
+//
+// -----------------------------------------------------------------------------
+///
+/// Extend the Message types with FieldMask utilities.
+///
+// -----------------------------------------------------------------------------
+
+import Foundation
+
+extension Message {
+
+  /// Checks whether the given path is valid for Message type.
+  ///
+  /// - Parameter path: Path to be checked
+  /// - Returns: Boolean determines path is valid.
+  public static func isPathValid(
+    _ path: String
+  ) -> Bool {
+    var message = Self()
+    return message.hasPath(path: path)
+  }
+
+  internal mutating func isPathValid(
+    _ path: String
+  ) -> Bool {
+    hasPath(path: path)
+  }
+}
+
+/// Defines available options for merging two messages.
+public struct MergeOption {
+
+  public init(replaceRepeatedFields: Bool = false) {
+    self.replaceRepeatedFields = replaceRepeatedFields
+  }
+
+  /// The default merging behavior will append entries from the source
+  /// repeated field to the destination repeated field. If you only want
+  /// to keep the entries from the source repeated field, set this flag
+  /// to true.
+  public var replaceRepeatedFields = false
+}
+
+extension Message {
+
+  /// Merges fields specified in a FieldMask into another message.
+  ///
+  /// - Parameters:
+  ///   - source: Message should be merged to the original one.
+  ///   - fieldMask: FieldMask specifies which fields should be merged.
+  public mutating func merge(
+    to source: Self,
```

Comment:
"to" seems wrong, it's this "from", i.e. - the `source` is where the fields are copied _from_?  Then are copied "to" `self`?